msal_extensions/token_cache.py

```diff
@@ -37,8 +38,13 @@ def _reload_if_necessary(self):
                 self._last_sync = time.time()
         except IOError as exp:
             if exp.errno != errno.ENOENT:
-                raise
-            # Otherwise, from cache's perspective, a nonexistent file is a NO-OP
+                if sys.platform.startswith('darwin'):
+                    from .osx import KeychainError
+                    if exp.exit_status != KeychainError.ITEM_NOT_FOUND:
+                        raise
+                else:
```

Comment:
This implementation does not address similar exceptions from other platforms, does it?

And, I have a feeling that, even after we complete all those platform-dependent logic here, this would not look as good as the #67 approach.

Feel free to explore more, and then we discuss.

Reply:
I remember closing the conversation on the fact that we just need to address the Keychain error for now and so for the time being we do it in the token cache layer. Similar errors in other platforms can be addressed by creating a new Exception at the persistence layer later. I remember the reason we do not want to introduce a None return value is because persistence layer is exposed as an API and so introducing a new exception at that layer is also something we possibly need to discuss. So eventually once we have had that conversation we can go ahead and visit #67. @henrik-me Am I missing something ?